Makefile

```diff
@@ -7,20 +7,32 @@ COVERAGE = python3-coverage
 
 DESTDIR=/
 
+.PHONY: all all-style all-fix \
+	build clean install \
+	check-all check check-tutorial check-nocompat check-compat \
+	check-pypy check-noextensions check-contrib \
+	typing style fix format reformat codespell \
+	coverage coverage-html doc sphinx apidocs
+
 all: build
 
-doc:: sphinx
+all-style:: style format codespell
 
-sphinx::
-	$(MAKE) -C docs html
+all-fix:: fix reformat codespell
 
 build::
 	$(SETUP) build
 	$(SETUP) build_ext -i
 
+clean::
+	$(SETUP) clean --all
+	rm -f dulwich/*.so
+
 install::
 	$(SETUP) install --root="$(DESTDIR)"
 
+check-all: check check-pypy check-noextensions
```

Comment:
This breaks when run with "make -j". 

"check-all" is a bit confusing name - what is this target really trying to do? "check" for most projects means running the test suite. I'm not sure why this also tests typing.

Reply:
I don't know :). This is an existing rule, I just moved it around. I agree that it doesn't seem to have a clear purpose.

Reply:
Let's just drop it?

Reply:
Can do.